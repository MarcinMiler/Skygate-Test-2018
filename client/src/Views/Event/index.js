import React from 'react'
import gql from 'graphql-tag'

import Container from '../../Components/Container'
import EventDetails from './Components/EventDetails'
import Map from './Components/Map'
import Buttons from './Components/Buttons'
import QueryWithLoading from '../../Components/QueryWithLoading'

const Event = ({ match }) => (
    <QueryWithLoading query={eventQuery} variables={{ id: match.params.id }}>
        {({ event }) => (
            <Container>
                <EventDetails event={event} />

                <Map event={event} />

                <Buttons />
            </Container>
        )}
    </QueryWithLoading>
)

const eventQuery = gql`
    query event($id: Int!) {
        event(id: $id) {
            id
            title
            description
            organizer
            location
            startDate
            endDate
            photo
            category
        }
    }
`

export default Event
