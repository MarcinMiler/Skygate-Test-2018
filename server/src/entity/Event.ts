import { Entity, PrimaryGeneratedColumn, Column, BaseEntity } from 'typeorm'

@Entity()
export class Event extends BaseEntity {
    @PrimaryGeneratedColumn() id: number

    @Column() title: string

    @Column() description: string

    @Column() organizer: string

    @Column() location: string

    @Column() startDate: string

    @Column() endDate: string

    @Column() photo: string

    @Column() category: string
}
