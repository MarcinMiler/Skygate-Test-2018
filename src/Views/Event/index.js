import React from 'react'
import Container from '../../Components/Container'
import EventDetails from './Components/EventDetails'
import Map from './Components/Map'

const Event = () => (
    <Container>
        <EventDetails />

        <Map />
    </Container>
)

export default Event
