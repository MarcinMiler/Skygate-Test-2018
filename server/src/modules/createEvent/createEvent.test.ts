import { Connection } from 'typeorm'
import { createTestConn } from '../../testUtils/createTestConn'
import { request } from 'graphql-request'
import { Event } from '../../entity/Event'

let conn: Connection
beforeAll(async () => {
    conn = await createTestConn(true)
})

afterAll(async () => {
    conn.close()
})

const url = process.env.TEST_HOST as string

const addEventMutation = `
    mutation {
        createEvent(
            title: "Color Festival"
            description: "Some event"
            organizer: "Somelab"
            location: "Warsaw"
            startDate: "11.07.2018 10:30"
            endDate: "11.07.2018 20:00"
            photo: "photo.jpeg"
            category: "Music") {
                id
                title,
                description,
                organizer,
                location,
                startDate,
                endDate,
                photo,
                category,
            }
    }
`

it('should create new event', async () => {
    const response = await request(url, addEventMutation)
    expect(response).toMatchSnapshot()

    const event = await Event.find()
    expect(event).toMatchSnapshot()
})
