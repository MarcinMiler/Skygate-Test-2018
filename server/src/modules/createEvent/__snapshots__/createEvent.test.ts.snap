// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create new event 1`] = `
Object {
  "createEvent": Object {
    "category": "Music",
    "description": "Some event",
    "endDate": "11.07.2018 20:00",
    "id": 1,
    "location": "Warsaw",
    "organizer": "Somelab",
    "photo": "photo.jpeg",
    "startDate": "11.07.2018 10:30",
    "title": "Color Festival",
  },
}
`;

exports[`should create new event 2`] = `
Array [
  Event {
    "category": "Music",
    "description": "Some event",
    "endDate": "11.07.2018 20:00",
    "id": 1,
    "location": "Warsaw",
    "organizer": "Somelab",
    "photo": "photo.jpeg",
    "startDate": "11.07.2018 10:30",
    "title": "Color Festival",
  },
]
`;
